Src/IronPython/Runtime/Operations/StringOps.cs

```diff
@@ -1870,13 +1868,15 @@ internal static Bytes DoEncodeUtf8(CodeContext context, string s)
             throw PythonOps.TypeError("'{0}' encoder returned '{1}' instead of 'bytes'; use codecs.encode() to encode to arbitrary types", encoding, DynamicHelpers.GetPythonType(res[0]).Name);
         }
 
-        private static string UserDecode(PythonTuple codecInfo, IList<byte> data, string errors) {
+        private static string UserDecode(PythonTuple codecInfo, object data, string errors) {
```

Comment:
Maybe `IBufferProtocol` instead of `object`? Unless you plan to call it from somewhere else.

Reply:
I was thinking of merging `StringOps.RawDecode` and `_codecs.decode`: they both seem to do the same, yet they have a different implementation. Right now, `UserDecode` is only used through `RawDecode`, which accepts only `IBufferProtocol`, but `_codecs.decode` accepts `object?`. So I wanted to keep it this way until it becomes clear what the final solution will be. I haven't looked into this issue deep enough yet.